### PR TITLE
Update outdated npm packages

### PR DESCRIPTION
**Pull request by pierlon**
## Summary

Closes #4857
Closes #4734

## Checklist

- [ ] My pull request is addressing an open issue (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -69,28 +95,8 @@ const blockEditor = {
 		'amp-block-editor': './assets/src/block-editor/index.js',
 		'amp-block-validation': './assets/src/block-validation/index.js',
 	},
-	module: {
```
Comment (pierlon) [inline webpack.config.js:72]: I've decided to use to the default CSS module loader rules. One effect of that change is that CSS file names will no longer have the name `[name]-compiled.css`, but simply `[name].css`. I've also updated the sections of code to reference the new file names.

File: webpack.config.js [hunk 2]
```diff
@@ -14,15 +13,44 @@ const WebpackBar = require( 'webpackbar' );
...
 		path: path.resolve( process.cwd(), 'assets', 'js' ),
 		filename: '[name].js',
 		chunkFilename: '[name].js',
 	},
+	module: {
+		...defaultConfig.module,
+		rules: [
+			...defaultConfig.module.rules.map(
+				( rule ) => {
+					// @todo Can remove once the default config no longer excludes excludes CSS related to Gutenberg components.
```
Comment (pierlon) [inline webpack.config.js:30]: The `setup` Webpack config fails when implicitly importing a CSS file from `node_modules/@wordpress/components/build-style/style.css`. I've modified the rule to include the `node_modules/@wordpress` directory for now, until the issue is resolved upstream.

I've opened https://github.com/WordPress/gutenberg/issues/23402 to track this.

**Review by westonruter — APPROVED** [review]
Looks good to me! But I defer to @johnwatkins0's experience to give the final approval.

**Review by johnwatkins0 — APPROVED** [review]
Looks good!

**Pull request merged**